azext_edge/edge/providers/adr/namespace_assets.py

```diff
@@ -0,0 +1,2121 @@
+            config_type="datapoint"
+        )
+        return datapoint
+
+    # otherwise process opcua specific configurations if provided
+    additional_configuration = {}
+    if queue_size is not None:
+        additional_configuration["queueSize"] = queue_size
+    if sampling_interval is not None:
+        additional_configuration["samplingInterval"] = sampling_interval
+    if additional_configuration:
+        from .specs import NAMESPACE_ASSET_OPCUA_DATAPOINT_CONFIGURATION_SCHEMA
+        ensure_schema_structure(
+            NAMESPACE_ASSET_OPCUA_DATAPOINT_CONFIGURATION_SCHEMA, input_data=additional_configuration
+        )
+    datapoint["dataPointConfiguration"] = json.dumps(additional_configuration)
+    # process configurations
+    return datapoint
+
+
+def _get_event(asset: dict, event_name: str) -> dict:
+    """Helper function to get an event from an asset.
+
+    Raises InvalidArgumentValueError if the event is not found.
+    """
+    events = asset["properties"].get("events", [])
+    matched_events = [event for event in events if event["name"] == event_name]
+    if not matched_events:
+        raise InvalidArgumentValueError(f"Event '{event_name}' not found in asset '{asset['name']}'.")
+    return matched_events[0]
+
+
+def _get_mgmt_group(asset: dict, management_group_name: str) -> dict:
+    """Helper function to get a management group from an asset.
+
+    Raises InvalidArgumentValueError if the management group is not found.
+    """
+    mgmt_groups = asset["properties"].get("managementGroups", [])
+    matched_mgmt_groups = [mgmt for mgmt in mgmt_groups if mgmt["name"] == management_group_name]
+    if not matched_mgmt_groups:
+        raise InvalidArgumentValueError(
+            f"Management group '{management_group_name}' not found in asset '{asset['name']}'."
+        )
+    return matched_mgmt_groups[0]
+
+
+# maybe move the config processing functions to specs?
+def _process_configs(
+    asset_type: str,
+    default: bool = True,
+    **kwargs
+) -> dict:
+    """Main function to process all of the config + destination args based on asset type.
+
+    Destination and custom configuration arguments will be treated as an overwrite rather than update.
+    For destinations, currently only one destination is supported but there may be more than one in the future.
+    """
+    result = {}
+    asset_type = asset_type.lower()
+    if asset_type == DeviceEndpointType.OPCUA.value.lower():
+        # allowed: datasets, events, mgmt groups, destinations must be mqtt
+        # not allowed: streams
+        # still waiting on opcua mgmt group schemas
+        result = {
+            "datasetsConfiguration": _process_opcua_dataset_configurations_v1(
+                **kwargs
+            ),
+            "eventsConfiguration": _process_opcua_event_configurations_v1(
+                **kwargs
+            ),
+            "managementGroupsConfiguration": process_additional_configuration(
```

Comment:
double check but I don't think these (mgmt for opcua + onvif) are being funneled through

Reply:
removed in #618